### PR TITLE
Fix circular dependency during Jest tests from importing constants from GarbageCollection 

### PR DESCRIPTION
**Pull request by NicholasCouri**
## Description

> Jets tests started failing when importing GC constant from BlobManager.cs, probably due to circular dependency when building their bundle. 

Additional information can be found: [Bug 2635](https://dev.azure.com/fluidframework/internal/_workitems/edit/2635)
Stack of interest:
Uncaught ReferenceError: Cannot access 'gcTreeKey' before initialization
    at Module.gcTreeKey (deltaScheduler.ts:33:48)
    at ../../../packages/runtime/container-runtime/lib/summaryFormat.js (summaryFormat.ts:180:100)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ../../../packages/runtime/container-runtime/lib/dataStoreContext.js (dataStore.ts:198:2)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)
    at ../../../packages/runtime/container-runtime/lib/dataStores.js (dataStoreRegistry.ts:34:2)
    at __webpack_require__ (bootstrap:24:1)
    at fn (hot module replacement:62:1)

### CODE REVIEW

File: packages/runtime/container-runtime/src/blobManager.ts [hunk 1]
```diff
@@ -158,10 +164,18 @@ export class BlobManager {
         private readonly runtime: IBlobManagerRuntime,
         stashedBlobs: IPendingBlobs = {},
     ) {
-        this.logger = ChildLogger.create(this.runtime.logger, "BlobManager");
+        this.mc = loggerToMonitoringContext(ChildLogger.create(this.runtime.logger, "BlobManager"));
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/blobManager.ts:167]: This code should not be part of this PR as this is a specific feature level work. Tyler is doing this in a separate PR.

**Comment by agarwal-navin** [comment]
FYI - This PR kind of does the same thing but I need to update it - https://github.com/microsoft/FluidFramework/pull/12282

File: packages/test/test-version-utils/src/compatConfig.ts [hunk 2]
```diff
@@ -30,8 +30,8 @@ interface CompatConfig {
     dataRuntime?: string | number;
 }
 
-// N, N - 1, and N - 2
```
Comment (agarwal-navin) [inline packages/test/test-version-utils/src/compatConfig.ts:33]: This change should not be here as well. I think you have unrrelated commits in your branch that you need to remove.

File: packages/runtime/container-runtime/src/blobManager.ts [hunk 3]
```diff
@@ -158,10 +164,18 @@ export class BlobManager {
...
+         // Read the feature flag that tells whether to throw when a tombstone blob is requested.
+         this.throwOnTombstoneUsage =
+         this.mc.config.getBoolean(throwOnTombstoneUsageKey) === true &&
+         this.runtime.clientDetails.type !== summarizerClientType;
+
         this.runtime.on("disconnected", () => this.onDisconnected());
         this.redirectTable = this.load(snapshot);
 
+        if (this.throwOnTombstoneUsage) {
+            console.log("test");
```
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/blobManager.ts:177]: Looks like an unintended `console.log`

File: packages/test/test-version-utils/src/compatConfig.ts (hunk 2 above)
Comment (NicholasCouri) [reply packages/test/test-version-utils/src/compatConfig.ts:33]: Yes, I will remove them - not sure how they got added

File: packages/runtime/container-runtime/src/blobManager.ts (hunk 1 above)
Comment (NicholasCouri) [reply packages/runtime/container-runtime/src/blobManager.ts:167]: I think Tyler has checked it in. Not sure why it shows as my changes.

**Comment by NicholasCouri** [comment]
> FYI - This PR kind of does the same thing but I need to update it - #12282

Oh, if you want to take over, I can close it.

File: packages/runtime/container-runtime/src/test/blobManager.spec.ts [hunk 4]
```diff
@@ -43,7 +43,11 @@ class NonDedupeStorage extends BaseMockBlobStorage {
 }
 
 class MockRuntime extends TypedEventEmitter<IContainerRuntimeEvents> implements IBlobManagerRuntime {
-    constructor(snapshot: IBlobManagerLoadInfo = {}, attached = false) {
+    constructor(
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/test/blobManager.spec.ts:46]: Please remove this. This is unrrelated.

**Review by agarwal-navin — APPROVED** [review]
